src/common/pico_base/BUILD.bazel

```diff
@@ -0,0 +1,97 @@
+package(default_visibility = ["//visibility:public"])
+
+genrule(
+    name = "version_header",
+    srcs = [
+        "//:pico_sdk_version.cmake",
+        "include/pico/version.h.in",
+    ],
+    outs = ["generated_include/pico/version.h"],
+    cmd = "$(location //bazel:parse_version) $(location //:pico_sdk_version.cmake)  --template=$(location include/pico/version.h.in) > $@",
+    cmd_bat = "$(location //bazel:parse_version) $(location //:pico_sdk_version.cmake)  --template=$(location include/pico/version.h.in) > $@",
+    tools = ["//bazel:parse_version"],
+    visibility = ["//visibility:private"],
+)
+
+cc_library(
+    name = "version",
+    hdrs = ["generated_include/pico/version.h"],
+    includes = ["generated_include"],
+)
+
+alias(
+    name = "platform_defs",
+    actual = select({
+        "//bazel/constraint:host": "//src/host/pico_platform:platform_defs",
+        "//bazel/constraint:rp2": "//src/rp2_common/pico_platform:platform_defs",
+    }),
+)
+
+alias(
+    name = "pico_platform",
+    actual = select({
+        "//bazel/constraint:host": "//src/host/pico_platform:pico_platform",
+        "//bazel/constraint:rp2": "//src/rp2_common/pico_platform:pico_platform",
+    }),
+)
+
+cc_library(
+    name = "pico_base_interface",
+    hdrs = [
+        "include/pico.h",
+        "include/pico/assert.h",
+        "include/pico/config.h",
+        "include/pico/error.h",
+        "include/pico/types.h",
+    ],
+    includes = ["include"],
+
+    # Be extra careful about who references this for now.
```

Comment:
Could the comment explain _why_ we need to be careful? What should the maintainers be carefully watching out for?